### PR TITLE
Added SiteContractServiceGraphListener structure to manage the Azure CNC listeners

### PR DESCRIPTION
**Pull request by sajagana**
Note:

Added SiteContractServiceGraphListener structure to schema_template_contract_service_graph.go, model.

### CODE REVIEW

File: models/schema_template_contract_service_graph.go [hunk 1]
```diff
@@ -65,3 +65,40 @@ func (graphAttr *SiteContractServiceGraph) ToMap() (map[string]interface{}, erro
 
 	return graphAttrMap, nil
 }
+
+type SiteContractServiceGraphListener struct {
```
Comment (akinross) [inline models/schema_template_contract_service_graph.go:69]: Why not leverage PatchPayloadList in model.go?

File: models/schema_template_contract_service_graph.go [hunk 2]
```diff
@@ -65,3 +65,40 @@ func (graphAttr *SiteContractServiceGraph) ToMap() (map[string]interface{}, erro
...
+	}
+
+	return &SiteContractServiceGraphListener{
+		Ops:   ops,
+		Path:  path,
+		Value: listenerMap,
+	}
+}
+
+func (listenerAttr *SiteContractServiceGraphListener) ToMap() (map[string]interface{}, error) {
```
Comment (akinross) [inline models/schema_template_contract_service_graph.go:96]: When PatchPayloadList is leverage it's ToMap() function can be used.

File: models/schema_template_contract_service_graph.go [hunk 3]
```diff
@@ -65,3 +65,40 @@ func (graphAttr *SiteContractServiceGraph) ToMap() (map[string]interface{}, erro
...
+			"nlbDevIp":     frontendIpDnMap,
+		}
+	}
+
+	return &SiteContractServiceGraphListener{
+		Ops:   ops,
+		Path:  path,
+		Value: listenerMap,
+	}
+}
```
Comment (akinross) [inline models/schema_template_contract_service_graph.go:94]:
```suggestion
func NewSiteContractServiceGraphListener(ops, path, name, protocol, securityPolicy string, port int, certificates []interface{}, rules []interface{}, frontendIpDnMap map[string]string) *PatchPayloadList {
	var listenerMap map[string]interface{}
	if ops != "remove" {
		listenerMap = map[string]interface{}{
			"name":         name,
			"port":         port,
			"protocol":     protocol,
			"secPolicy":    securityPolicy,
			"certificates": certificates,
			"rules":        rules,
			"nlbDevIp":     frontendIpDnMap,
		}
	}

	return &PatchPayloadList{
		Ops:   ops,
		Path:  path,
		Value: listenerMap,
	}
}
```

File: models/schema_template_contract_service_graph.go [hunk 4]
```diff
@@ -65,3 +65,40 @@ func (graphAttr *SiteContractServiceGraph) ToMap() (map[string]interface{}, erro
 
 	return graphAttrMap, nil
 }
+
+type SiteContractServiceGraphListener struct {
+	Ops   string                 `json:",omitempty"`
+	Path  string                 `json:",omitempty"`
+	Value map[string]interface{} `json:",omitempty"`
+}
```
Comment (akinross) [inline models/schema_template_contract_service_graph.go:73]:
```suggestion
```

File: models/schema_template_contract_service_graph.go [hunk 5]
```diff
@@ -65,3 +65,40 @@ func (graphAttr *SiteContractServiceGraph) ToMap() (map[string]interface{}, erro
...
+
+func (listenerAttr *SiteContractServiceGraphListener) ToMap() (map[string]interface{}, error) {
+	listenerAttrMap := make(map[string]interface{})
+	A(listenerAttrMap, "op", listenerAttr.Ops)
+	A(listenerAttrMap, "path", listenerAttr.Path)
+	if listenerAttr.Value != nil {
+		A(listenerAttrMap, "value", listenerAttr.Value)
+	}
+	return listenerAttrMap, nil
+}
```
Comment (akinross) [inline models/schema_template_contract_service_graph.go:104]:
```suggestion
```

**Review by akinross — CHANGES_REQUESTED** [review]

File: models/schema_template_contract_service_graph.go [hunk 6]
```diff
@@ -65,3 +65,40 @@ func (graphAttr *SiteContractServiceGraph) ToMap() (map[string]interface{}, erro
...
+
+type SiteContractServiceGraphListener struct {
+	Ops   string                 `json:",omitempty"`
+	Path  string                 `json:",omitempty"`
+	Value map[string]interface{} `json:",omitempty"`
+}
+
+func NewSiteContractServiceGraphListener(ops, path, name, protocol, securityPolicy string, port int, certificates []interface{}, rules []interface{}, frontendIpDnMap map[string]string) *SiteContractServiceGraphListener {
+	var listenerMap map[string]interface{}
+	if ops != "remove" {
```
Comment (akinross) [inline models/schema_template_contract_service_graph.go:77]: I don't think this check is needed when you will use, GetRemovePatchPayload in the resource.

**Review by akinross — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**